compiler/rustc_query_impl/src/on_disk_cache.rs

```diff
@@ -244,7 +246,26 @@ impl<'sess> rustc_middle::ty::OnDiskCache<'sess> for OnDiskCache<'sess> {
         }
     }
 
-    fn serialize(&self, tcx: TyCtxt<'sess>, encoder: &mut FileEncoder) -> FileEncodeResult {
+    fn drop_serialized_data(&self, tcx: TyCtxt<'tcx>) {
```

Comment:
Is there a reason why this is in its own method instead of being part of `serialize`?

Reply:
This function invokes queries, so needs the on-disk cache to be available. Meanwhile, references to the mmapped file need to be dropped before calling `save_in` which removes the backing file. I will add a comment.